scheduler/core/programprovider/gpp/gppprogramprovider.py

```diff
@@ -739,138 +736,47 @@ def find_filter(filter_input: str, filter_dict: Mapping[str, float]) -> Optional
-                            GppProgramProvider._fpu_to_barcode[instrument][fpu], description=fpu)
-                            for fpu in fpus])
-                        disperser_resources = frozenset(
-                            [self._sources.origin.resource.lookup_resource(disperser.split('_')[0],
-                                                                           resource_type=ResourceType.DISPERSER)
-                             for disperser in dispersers])
-                    resources = frozenset([r for r in fpu_resources | disperser_resources | instrument_resources])
-
-                    # Close previous atom, if any
-                    if n_atom > 0:
-                        mode = determine_mode(instrument)
-                        previous_atom = atoms[-1]
-                        previous_atom.qa_state = min(qa_states, default=QAState.NONE)
-                        if previous_atom.qa_state is not QAState.NONE:
-                            previous_atom.observed = True
-                        previous_atom.resources = resources
-                        previous_atom.guide_state = any(guiding)
-                        previous_atom.wavelengths = frozenset(wavelengths)
-                        previous_atom.obs_mode = mode
-
-                    # New atom entry
-                    n_atom += 1
-
-                    classes = []
-                    guiding = []
-                    resources = []
-                    wavelengths = []
-                    fpus = []
-                    dispersers = []
-                    filters = []
-                    mode = None
-                    atoms.append(Atom(id=atom_id,
-                                      exec_time=ZeroTime,
-                                      prog_time=ZeroTime,
-                                      part_time=ZeroTime,
-                                      program_used=ZeroTime,
-                                      partner_used=ZeroTime,
-                                      not_charged=ZeroTime,
-                                      observed=False,
-                                      qa_state=QAState.NONE,
-                                      guide_state=False,
-                                      resources=resources,
-                                      wavelengths=frozenset(wavelengths),
-                                      obs_mode=mode))
-                    prev_atom_id = atoms[-1].id
-
-                # If FPU is None, 'None', or FPU_NONE, which are effectively the same thing, we ignore.
-                if fpu is not None and fpu != 'None' and fpu != 'FPU_NONE':
-                    fpus.append(fpu)
-                dispersers.append(disperser)
-                if filt and filt != 'None':
-                    filters.append(filt)
-                wavelengths.append(wavelength)
+                step_time = int(step[GppProgramProvider._AtomKeys.TOTAL_TIME]) // 1000000 # transform to seconds
```

Comment:
MICROSECONDS_PER_SECOND = 1000000
step_time = int(step[GppProgramProvider._AtomKeys.TOTAL_TIME]) // MICROSECONDS_PER_SECON

Just to avoid magic numbers :)